src/io/aleph/dirigiste/Pool.java

```diff
@@ -406,19 +408,17 @@ public void run() {
     public void acquire(final K key, final AcquireCallback<V> callback) {
         final long start = System.nanoTime();
 
+        _lock.lock();
```

Comment:
We need to acquire the lock sooner to ensure the `startControlLoop` won't destroy
our `Pool` in which we are currently generating a new object.